### PR TITLE
[Fizz] Postponing in the shell

### PR DESCRIPTION
**Pull request by sebmarkbage**
When we postpone a prerender in the shell, we should just leave an empty prelude and resume from the root. While preserving any options passed in.

Since we haven't flushed anything we can't assume we've already emitted html/body tags or any resources tracked in the resumable state. This introduces a resetResumableState function to reset anything we didn't flush.

This is a bit hacky. Ideally, we probably shouldn't have tracked it as already happened until it flushed or something like that.

Basically, it's like restarting the prerender with the same options and then immediately aborting. When we add the preload headers, we'd track those as preload() being emitted after the reset and so they get readded to the resumable state in that case.

### CODE REVIEW

File: packages/react-server/src/ReactFizzServer.js [hunk 1]
```diff
@@ -2477,6 +2511,17 @@ function trackPostpone(
...
+  if (boundary === null) {
+    segment.id = request.nextSegmentId++;
+    trackedPostpones.rootSlots = segment.id;
+    if (request.completedRootSegment !== null) {
+      // Postpone the root if this was a deeper segment.
+      request.completedRootSegment.status = POSTPONED;
+    }
+    return;
+  }
+
```
Comment (gnoff) [inline packages/react-server/src/ReactFizzServer.js:2524]: There is a reference below to
```
throw new Error(
  'It should not be possible to possible to postpone at the root...
 )
```

This whole boundaryKeyPath existence check can go away or if you're concerned it could still happen in maybe we just make the message clear it isn't about postponing at the root

**Review by gnoff — APPROVED** [review]

File: packages/react-server/src/ReactFizzServer.js (hunk 1 above)
Comment (sebmarkbage) [reply packages/react-server/src/ReactFizzServer.js:2524]: The "root" in that case means the very first React node before entering the first element.

I'm not sure if that's actually impossible. I think it might be if you pass in a `React.lazy` at the root which then postpones in the reject.

Comment (gnoff) [reply packages/react-server/src/ReactFizzServer.js:2524]: ah ty
